File: Week1/Day1/AppleStocks.java
public class AppleStocks {


    public static int getMax(int[] arr)
    {
        int min=arr[0];
        int ans=Integer.MIN_VALUE;
        for(int i=1;i<arr.length;i++)
        {
            int max=arr[i]-min;
            min=Math.min(min,arr[i]);
            if(max>ans)
            {
                ans=max;
            }
        }
        return ans;
    }




    public static void main(String[] args) {
        int[] stockPricesYesterday = {10, 7, 5, 8, 11, 9};
        int[] case1={1,2,3,4};
        int[] case2={100, 200, 300, 200, 100, 50, 200, 500};
        int[] case3={4, 3, 2, 1};
        int[] case4={100,200,300,200,100,50};



        System.out.println(getMax(stockPricesYesterday));
        System.out.println(getMax(case1));
        System.out.println(getMax(case2));
        System.out.println(getMax(case3));
        System.out.println(getMax(case4));


    }
}

File: Week1/Day1/AppleStocks.py
def getMax(arr):
    if len(arr)<=1:
        return False;
    elif(len(arr)==2):
        return arr[1]-arr[0]
    else:
        minimum=arr[0]
        max_profit=arr[1]-arr[0]
        for i in range(2,len(arr)):
            minimum=min(minimum,arr[i-1])
            max=arr[i]-minimum
            if max>max_profit:
                max_profit=max
        return max_profit


if __name__=="__main__":
    stocks=[10,7,5,8,11,9]
    stocks1=[1,2,3,4]
    stocks2=[100, 200, 300, 200, 100, 50, 200, 500]
    stocks3=[4, 3, 2, 1]
    print(getMax(stocks))
    print(getMax(stocks1))
    print(getMax(stocks2))
    print(getMax(stocks3))




